edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -350,13 +354,32 @@ impl ModuleRuntime for DockerModuleRuntime {
                     module.config().image()
                 );
 
-                let create_options = create_options
+                let mut create_options = create_options
                     .with_image(module.config().image().to_string())
                     .with_env(merged_env)
                     .with_labels(labels);
 
-                // Here we don't add the container to the iot edge docker network as the edge-agent is expected to do that.
-                // It contains the logic to add a container to the iot edge network only if a network is not already specified.
+                // In the bootstrap scenario (edge runtime module starting up for the first time),
+                // add the container to the custom network. We identify this scenario by looking
+                // for 'NetworkId' in the module's environment. The edge runtime module is then
+                // responsible for adding all subsequent modules to the custom network.
+                if let Some(network_id) = module.env().get("NetworkId") {
```

Comment:
Just to clarify, `NetworkId` will actually always be defined, right? All user modules created by Edge Agent will have it set by Edge Agent already, and now with this change Edge Agent will have it set by the watchdog?

Reply:
I _think_ that because of the merge of environments (env from create options + env from spec) earlier in the function, most if not all modules will have `NetworkId` defined, but only edge agent in the bootstrap case will have it defined in the spec. I'm specifically looking in the spec (`module.env()`) here, not `merged_env`. But I may not be following the logic correctly...what do you think?

Reply:
Ah yes, I forgot there are two different env maps.